examples/websocket-relay/relay-app/src/lib/workflow.ts

```diff
@@ -132,90 +158,83 @@ export function fail(workflowId: WorkflowId) {
+  // @ts-ignore-next-line
+  const status = data.result.metadata.replayed ? "replayed" : "executed";
+  // @ts-ignore-next-line
+  const receipt = parseReceipt(data.result.receipt);
 
-    if (!activeWorkflow) {
-      console.error("Received a receipt but workflow was not initiated");
-      return;
-    }
+  // Update task in UI
+  taskStore.update((store) => {
+    const updatedTasks = store[activeWorkflow.id].map((t) =>
+      t.id === taskId
+        ? {
+            ...t,
+            status,
+            message: getTaskMessage(status),
+            receipt,
+          }
+        : t
+    );
 
-    const taskId = activeWorkflow.step + 1;
-    const status = message.metadata.replayed ? "replayed" : "executed";
-    const receipt = parseReceipt(message.receipt);
-
-    // Update task in UI
-    taskStore.update((store) => {
-      const updatedTasks = store[activeWorkflow.id].map((t) =>
-        t.id === taskId
-          ? {
-              ...t,
-              status,
-              message: getTaskMessage(status),
-              receipt,
-            }
-          : t
-      );
-
-      return { ...store, [activeWorkflow.id]: updatedTasks };
-    });
+    return { ...store, [activeWorkflow.id]: updatedTasks };
+  });
 
-    // Log receipt
-    console.table(receipt);
+  // Log receipt
+  console.table(receipt);
 
-    if (activeWorkflow.step === activeWorkflow.tasks.length - 1) {
-      // Workflow is done. Reset workflow status to waiting.
-      workflowStore.update((workflows) => ({
-        ...workflows,
-        [activeWorkflow.id]: {
-          ...workflows[activeWorkflow.id],
-          status: "waiting",
-        },
-      }));
+  if (activeWorkflow.step === activeWorkflow.tasks.length - 1) {
+    // Workflow is done. Reset workflow status to waiting.
+    workflowStore.update((workflows) => ({
+      ...workflows,
+      [activeWorkflow.id]: {
+        ...workflows[activeWorkflow.id],
+        status: "waiting",
+      },
+    }));
 
-      // Deactivate workflow
-      activeWorkflowStore.set(null);
-    } else {
-      // Increment workflow step
-      activeWorkflowStore.update((store) =>
-        store ? { ...store, step: store.step + 1 } : null
-      );
-    }
+    // Deactivate workflow
+    activeWorkflowStore.set(null);
   } else {
-    console.warn("Received an unexpected message", message);
+    // Increment workflow step
+    activeWorkflowStore.update((store) =>
+      store ? { ...store, step: store.step + 1 } : null
+    );
   }
 }
 
-function parseReceipt(raw: {
+type RawReceipt = {
   iss: string | null;
   meta: Meta | null;
-  out: ["ok" | "error", Record<"/", Record<"bytes", string>>];
+  out: ["ok" | "error", Uint8Array];
   prf: string[];
   ran: Record<"/", string>;
-}): Receipt {
+};
+
+const parseReceipt = (raw: RawReceipt): Receipt => {
   return {
     iss: raw.iss,
     meta: raw.meta,
-    out: [raw.out[0], raw.out[1]["/"].bytes],
+    out: [raw.out[0], base64.encode(raw.out[1])],
```

Comment:
why are you base encoding this ? its ui related?

Reply:
Yes, we are using it to display the images here:

https://github.com/ipvm-wg/homestar/blob/ebc8fb1595ed01eb7123d69d6503bde2dd98ac3a/examples/websocket-relay/relay-app/src/components/node/Node.svelte#L59